### PR TITLE
Verifier interface 

### PR DESCRIPTION
**Pull request by edmundnoble**
This is an interface for "verifiers", plug-ins into tx verification that are able to grant capabilities and potentially even modify transactions.

This relies on the Pact branch also named edmund/spi. 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205797981912170

### CODE REVIEW

**Comment by edmundnoble** [comment]
The Pact branch has been merged.

File: src/Chainweb/Pact/TransactionExec.hs [hunk 1]
```diff
@@ -876,8 +937,9 @@ buyGas isPactBackCompatV16 cmd (Miner mid mks) = go
           interp mc = Interpreter $ \_input ->
             put (initState mc logGas) >> run (pure <$> eval buyGasTerm)
 
+      -- no verifiers are allowed in buy gas
       result <- applyExec' 0 (interp mcache) buyGasCmd
-        (_pSigners $ _cmdPayload cmd) bgHash managedNamespacePolicy
+        (_pSigners $ _cmdPayload cmd) [] bgHash managedNamespacePolicy
```
Comment (emilypi) [inline src/Chainweb/Pact/TransactionExec.hs:945]: 👍

File: src/Chainweb/VerifierPlugin/Allow.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{-# language BangPatterns #-}
+{-# language ScopedTypeVariables #-}
+{-# language OverloadedStrings #-}
+{-# language FlexibleContexts #-}
+module Chainweb.VerifierPlugin.Allow(plugin) where
```
Comment (emilypi) [inline src/Chainweb/VerifierPlugin/Allow.hs:5]: This is diagnostic test code only

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs [hunk 3]
```diff
@@ -1192,6 +1196,58 @@ pact410UpgradeTest = do
...
+          ]
+        )
+        (mkExec' "(free.m.x)")
+      )
+      (\cr -> liftIO $ do
+        assertTxSuccess "should have succeeded" (pDecimal 1) cr
+        -- The **Allow** verifier costs 100 gas flat
+        assertEqual "gas should have been charged" 344 (_crGas cr)
+      )
+    ]
```
Comment (emilypi) [inline test/Chainweb/Test/Pact/PactMultiChainTest.hs:1250]: 👍

**Review by emilypi — COMMENTED** [review]
Looking good. Some comments from our joint review.

**Review by emilypi — APPROVED** [review]
LOOKS. FING. GOOD.

**Review by chessai — APPROVED** [review]

**Pull request merged**